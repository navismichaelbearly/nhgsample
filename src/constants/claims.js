export const permissions = 'http://nhhg.org.uk/claims/permissions';
export const userType = 'http://nhhg.org.uk/claims/user_type';
export const roles = 'http://nhhg.org.uk/claims/roles';
