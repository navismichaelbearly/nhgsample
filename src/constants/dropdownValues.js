export const TASKS_SEL = 'Tasks';
export const ARREARS_SEL = 'Arrears';
export const PRIORITY_SEL = 'priority';
export const CURRENT_BALANCE_SEL = 'currentBalance';
export const DAYS_IN_ARREARS_SEL = 'daysInArrears';
export const MY_TASKS_SEL = 'myTasks';
export const USER_TASKS_SEL = 'selectedUserTasks';
