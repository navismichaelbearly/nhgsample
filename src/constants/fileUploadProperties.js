const properties = Object.freeze({
  acceptedFileExtensions:
    '.doc,.docx,.xls,.xlsx,.ppt,.pptx,.pdf,.txt,.jpg,.png,.rtf,.htm,.html,.msg,.eml,.mht,.ics,.csv,.oft',
  maxFileSizeBytes: 1024 * 1024 * 20, // 20mb
  maxFilesDefault: 10,
  maxFilesNotes: 1,
});

export default properties;
