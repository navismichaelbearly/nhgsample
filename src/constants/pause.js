// original statuses
export const OPEN = 'Open';
export const SCHEDULED = 'Scheduled';
export const CANCELLED = 'Cancelled';
export const COMPLETED = 'Completed';
// statuscodes
export const OPENSTATUSCODE = 1;
export const SCHEDULEDSTATUSCODE = 4;
export const CANCELLEDSTATUSCODE = 3;
export const COMPLETEDSTATUSCODE = 2;
export const UPDATESTATUSCODE = 99;

// mapped statuses
export const DRAFT = 'Draft';
export const ACTIVE = 'Active';

// activity type
export const PAUSE = 'Pause';
