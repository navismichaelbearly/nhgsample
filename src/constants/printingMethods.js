const LOCALPRINT = 'Local_Print';

export default LOCALPRINT;
