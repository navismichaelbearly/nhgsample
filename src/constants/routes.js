import devElse from '../util/devElse';
import {
  activitiesApi as activitiesApiBase,
  arrearsApi as arrearsApiBase,
  arrearsApiV2 as arrearsApiBaseV2,
  avatarsApi as avatarsApiBase,
  correspondenceApi as correspondenceApiBase,
  customerDashboard as customerDashboardBase,
  customersApi as customersApiBase,
  documentsApi as documentsApiBase,
  employeeDashboard as employeeDashboardBase,
  interactionsApi as interactionsApiBase,
  linkedCasesApi as linkedCasesApiBase,
  notesApi as notesApiBase,
  paymentsApi as paymentsApiBase,
  paymentsBaseURL as paymentsBase,
  tasksApi as tasksApiBase,
  tasksApiV2,
  sharepointApi as sharepointApiBase,
  identityApi as identityApiBase,
} from '../constants/tokens';

export const activitiesApi = devElse(
  'https://mynottinghill-dev.workwise.london/activities/web/api/v1',
  activitiesApiBase
);

export const arrearsApi = devElse(
  'https://mynottinghill-dev.workwise.london/arrears/web/api/v1',
  arrearsApiBase
);
export const arrearsApiV2 = devElse(
  'https://mynottinghill-dev.workwise.london/arrears/web/api/v2',
  arrearsApiBaseV2
);
export const identityApi = devElse(
  'https://mynottinghill-dev.workwise.london/identity/web/api/v1/',
  identityApiBase
);

export const avatarsApi = devElse(
  'https://nhhuksmedsamediad.blob.core.windows.net/employees/$id/Image/profile.jpg/profile_original.jpg',
  avatarsApiBase
);

export const correspondenceApi = devElse(
  'https://mynottinghill-dev.workwise.london/correspondence/web/api/v1/',
  correspondenceApiBase
);

export const customersApi = devElse(
  'https://mynottinghill-dev.workwise.london/customers/web/api/v1',
  customersApiBase
);

export const employeeDashboard = devElse(
  'https://mynottinghill-dev.workwise.london/dashboards/employees',
  employeeDashboardBase
);

export const employeeDashboardSearchResults = `${employeeDashboard}/search-results`;

export const customerDashboard = devElse(
  'https://mynottinghill-dev.workwise.london/dashboards/customers',
  customerDashboardBase
);

export const interactionsApi = devElse(
  'https://mynottinghill-dev.workwise.london/interactions/web/api/v1',
  interactionsApiBase
);

export const linkedCasesApi = devElse(
  'https://mynottinghill-dev.workwise.london/linkedcases/web/api/v1',
  linkedCasesApiBase
);

export const notesApi = devElse(
  'https://mynottinghill-dev.workwise.london/notes/web/api/v1',
  notesApiBase
);

export const payments = devElse(
  'https://mynottinghill-dev.workwise.london/payments/web',
  paymentsBase
);

export const paymentsApi = devElse(
  'https://mynottinghill-dev.workwise.london/payments/web/api/v1',
  paymentsApiBase
);

export const tasksApi = devElse(
  'https://mynottinghill-dev.workwise.london/tasks/web/api/v1',
  tasksApiBase
);

export const tasksApiRouteV2 = devElse(
  'https://mynottinghill-dev.workwise.london/tasks/web/api/v2',
  tasksApiV2
);

export const documentsApi = devElse(
  'https://mynottinghill-dev.workwise.london/documents/web/api/v1',
  documentsApiBase
);

export const sharepointApi = devElse(
  'https://mynottinghill-dev.workwise.london/documents/sharepoint',
  sharepointApiBase
);

export const dashboardDetailsForCustomer = customerId =>
  `${employeeDashboard}/customer/${customerId}`;
