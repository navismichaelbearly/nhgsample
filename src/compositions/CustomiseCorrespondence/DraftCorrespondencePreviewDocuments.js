import { Loader, Typography } from 'nhh-styles';
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import { PreviewLinkContainer, PreviewLink, PreviewTemplate } from '../../components';
import { errorTextLabels, labelsPropTypes } from './types';
import forcePdfDownload from '../../util/forcePdfDownload';
import {
  PDF as ENCODING_TYPE_PDF,
  TEXT as ENCODING_TYPE_TEXT,
} from '../../constants/documentEncodingTypes';
import { HTML, TEXT } from '../../constants/previewTemplateTypes';
import { FailedFatal, FailedRetriable } from '../../constants/correspondenceErrorTypes';
import { Heading3, P, InLineLightP } from '../../components/Common/styled';

const { Error, H5 } = Typography;

class DraftCorrespondencePreviewDocuments extends React.PureComponent {
  constructor(props) {
    super(props);
    const documentIds = this.props.documents.reduce((acc, document) => {
      acc[document.id] = false;
      return acc;
    }, {});

    this.state = {
      ...documentIds,
    };
  }

  getPreviews(document) {
    const {
      labels: { viewPreview, alreadyPreviewed, manualPrint },
      dataBddPrefix,
      isLocalPrint,
    } = this.props;

    const { id, type, sendingMethod, errorType, recipientName } = document;
    const isPostalLetter =
      isLocalPrint && type.toLowerCase() === 'letter' && sendingMethod.toLowerCase() === 'post';

    return (
      <Fragment>
        <H5>
          {recipientName} ({sendingMethod})
        </H5>
        <PreviewLinkContainer>
          {errorType ? (
            <Error>{this.documentError(errorType)}</Error>
          ) : (
            <Fragment>
              {isPostalLetter && (
                <P className="m-0" data-bdd="manual-print">
                  {manualPrint}
                </P>
              )}
              <PreviewLink
                data-state-key={'previewLink'}
                data-bdd={`${dataBddPrefix}-previewLink`}
                onClick={() => this.handleDocumentPreview(document)}
                isText
              >
                {viewPreview}
              </PreviewLink>
              {this.state[id] === true && (
                <InLineLightP data-bdd="document-viewed">{alreadyPreviewed}</InLineLightP>
              )}
            </Fragment>
          )}
        </PreviewLinkContainer>
      </Fragment>
    );
  }

  /**
   * takes an id of a document and update state to show that the
   * user has clicked the preview link
   */
  markDocumentViewed(id) {
    this.setState({
      [id]: true,
    });
  }

  documentError = errorType => {
    let message = null;
    if (errorType === FailedFatal) {
      message = this.props.errorText.fatalError;
    } else if (errorType === FailedRetriable) {
      message = this.props.errorText.retriableError;
    }
    return message;
  };

  generationError = () => {
    let message = null;
    if (this.props.fatalError) {
      message = this.props.errorText.fatalError;
    } else if (this.props.retriableError) {
      message = this.props.errorText.retriableError;
    }
    return message;
  };

  handleDocumentPreview = ({ content, encodingType, id }) => {
    switch (encodingType) {
      case ENCODING_TYPE_PDF:
        forcePdfDownload(content);
        this.props.onPreview(id);
        break;
      case ENCODING_TYPE_TEXT:
      case HTML:
        {
          const template = (
            <PreviewTemplate
              content={content}
              dataBddPrefix={`draftCorrespondence-template`}
              onClose={this.props.closeTemplate}
              errorMessage={this.props.errorText.noTemplateFound}
              label={this.props.labels.closeTemplate}
              type={encodingType === HTML ? HTML : TEXT}
            />
          );
          this.props.onPreview(id, { template, limitMaxWidth: true });
        }
        break;
      default:
    }

    this.markDocumentViewed(id);
  };

  render() {
    const {
      labels: { generatedCorrespondence, viewPreviewsBeforeProceeding },
      documents,
    } = this.props;

    if (this.props.generatePreviewLoading) {
      return <Loader />;
    }

    return (
      <Fragment>
        {/* eslint-disable no-nested-ternary */}
        {documents.length ? (
          <Fragment>
            <Heading3>{generatedCorrespondence}</Heading3>
            <P>{viewPreviewsBeforeProceeding}</P>
            {documents.map(document => (
              <Fragment key={document.id}>
                {document.loading ? <Loader /> : this.getPreviews(document)}
              </Fragment>
            ))}
          </Fragment>
        ) : this.generationError() ? (
          <Error>{this.generationError()}</Error>
        ) : null}
      </Fragment>
    );
  }
}

DraftCorrespondencePreviewDocuments.defaultProps = {
  documents: [],
  fatalError: false,
  generatePreviewLoading: false,
  retriableError: false,
  isLocalPrint: false,
};

DraftCorrespondencePreviewDocuments.propTypes = {
  closeTemplate: PropTypes.func.isRequired,
  dataBddPrefix: PropTypes.string.isRequired,
  errorText: errorTextLabels.isRequired,
  labels: labelsPropTypes.isRequired,
  onPreview: PropTypes.func.isRequired,
  documents: PropTypes.arrayOf(
    PropTypes.shape({
      loading: PropTypes.bool.isRequired,
      content: PropTypes.string,
      encodingType: PropTypes.string,
      errorMessages: PropTypes.object,
      errorType: PropTypes.string,
      id: PropTypes.string,
      recipientName: PropTypes.string,
      sendingMethod: PropTypes.string,
      status: PropTypes.string,
      type: PropTypes.string,
    }).isRequired
  ),
  fatalError: PropTypes.bool,
  generatePreviewLoading: PropTypes.bool,
  isLocalPrint: PropTypes.bool,
  retriableError: PropTypes.bool,
};

export default DraftCorrespondencePreviewDocuments;
