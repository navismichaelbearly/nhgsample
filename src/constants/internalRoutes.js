import { ACTIVE, DRAFT } from './pause';
import { isUserSeniorManager } from '../util/userRoles';
import legalReferralStatus, { status as legalReferralCaseStatus } from './legalReferral';

export const getActivityHistoryItemRoute = (arrearsId, itemId) =>
  `/arrears-details/${arrearsId || ':arrearsId'}/history/${itemId || ':itemId'}`;

export const getArrearsDetailRoute = arrearsId => `/arrears-details/${arrearsId || ':arrearsId'}`;

export const getArchivedTasksRoute = arrearsId =>
  `/arrears-details/${arrearsId || ':arrearsId'}/archived-tasks`;

export const getLegalReferralLink = ({
  arrearsId,
  legalReferralId,
  canCreateLegalReferral,
  status,
  userRoles,
}) => {
  const noLink = () => '';
  // pre-condition
  if (!arrearsId) {
    return noLink();
  }
  // helpers
  const prependedURL = `/arrears-details/${arrearsId}`;
  const reviewLink = () => `${prependedURL}/legal-case-referral/${legalReferralId}/review`;
  const approveLink = () => `${prependedURL}/legal-case-referral/${legalReferralId}/approve`;
  const createLink = () => `${prependedURL}/legal-case-referral/create`;
  const isLoggedUserSeniorManager = isUserSeniorManager(userRoles);
  // Legal Referral Task
  if (!legalReferralId) {
    if (canCreateLegalReferral && !isLoggedUserSeniorManager) {
      return createLink();
    }
    return noLink();
  }
  // Owner Sign-Off Task
  if (status === legalReferralCaseStatus.pendingSubmission) {
    if (isLoggedUserSeniorManager) {
      return approveLink();
    }
    return reviewLink();
  }
  // Manager Sign-Off
  if (status === legalReferralCaseStatus.requiresApproval) {
    return approveLink();
  }
  return reviewLink();
};

export const getTaskActivityRouteByActionType = ({
  actionType,
  arrearsId,
  legalReferral = {
    legalReferralId: '',
    canCreateLegalReferral: false,
    status: '',
    userRoles: [],
  },
}) => {
  const prependedURL = `/arrears-details/${arrearsId}`;
  switch (actionType) {
    case 'addinteraction': {
      return `${prependedURL}/interaction/create`;
    }
    case 'addnote': {
      return `${prependedURL}/note/create`;
    }
    case 'sendcorrespondence': {
      return `${prependedURL}/send-correspondence/create`;
    }
    case 'legalreferral': {
      return getLegalReferralLink({
        arrearsId,
        legalReferralId: legalReferral.legalReferralId,
        canCreateLegalReferral: legalReferral.canCreateLegalReferral,
        status: legalReferral.status,
        userRoles: legalReferral.userRoles,
      });
    }
    default: {
      return '#';
    }
  }
};

export default (
  { id: arrearsId, legalReferral: lr, legalReferralCase, pauseSummary, paymentPlan: pp },
  isManager
) => {
  // paymentPlan routing conditions
  const paymentPlanId = pp ? pp.id : null;

  // pause routing conditions
  const pauses = pauseSummary || [];
  const pauseAwaitingApproval = pauses.find(pause => pause.status === DRAFT);
  const activePause = pauses.find(pause => pause.status === ACTIVE);
  const currentPause = pauseAwaitingApproval || activePause || {};
  let pauseAction;
  if (currentPause.id) pauseAction = 'update';
  if (pauseAwaitingApproval && isManager) pauseAction = 'approve';

  // legalReferral routing conditions
  const legalReferralActionStatus = legalReferralCase ? legalReferralCase.status : null;
  const currentLegalReferralAction = lr ? lr.caseId : null;
  const legalReferralActionType =
    legalReferralActionStatus === legalReferralStatus.requiresApproval ? 'approve' : 'review';

  const linkToPage = (type, typeId, action) =>
    `/arrears-details/${arrearsId}/${type}/${typeId || 'create'}${
      typeId && action ? `/${action}` : ''
    }`;

  return {
    addNote: linkToPage('note'),
    addTask: linkToPage('task'),
    arrearsDetailNOSPModal: `/arrears-details/${arrearsId}?modal=serveNOSP`,
    interaction: linkToPage('interaction'),
    legalReferral: linkToPage(
      'legal-case-referral',
      currentLegalReferralAction,
      legalReferralActionType
    ),
    pause: linkToPage('pause-arrears-case', currentPause.id, pauseAction),
    paymentPlan: linkToPage('payment-plan', paymentPlanId),
    sendCorrespondence: linkToPage('send-correspondence'),
    tenancySustainmentReferral: linkToPage('refer-arrears-case'),
    welfareReformDetails: `/arrears-details/${arrearsId}/welfare-reform-details`,
  };
};
