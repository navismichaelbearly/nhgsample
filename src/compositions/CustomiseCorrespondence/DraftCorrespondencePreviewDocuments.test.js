import { Loader } from 'nhh-styles';
import path from 'ramda/src/path';
import React from 'react';
import { shallow } from 'enzyme';
import Dictionary from '../../constants/dictionary';
import DraftCorrespondencePreviewDocuments from './DraftCorrespondencePreviewDocuments';
import forcePdfDownload from '../../util/forcePdfDownload';
import { FailedFatal, FailedRetriable } from '../../constants/correspondenceErrorTypes';
import { HTML, TEXT } from '../../constants/documentEncodingTypes';

jest.mock('../../util/forcePdfDownload');

const dictionary = Dictionary();
function sel(id) {
  return `[data-state-key="${id}"]`;
}

const defaultProps = {
  closeTemplate: () => {},
  dataBddPrefix: '',
  documents: [
    {
      content: 'JVBERi0xLjUNCjQgMCBvYmoNCjw8L1R5cGUgL1BhZ2UvUGFyZW',
      encodingType: 'Pdf',
      errorMessages: { failures: null },
      erroType: null,
      id: 'asdasd213123',
      loading: false,
      recipientName: 'recipient name',
      sendingMethod: 'Post',
      status: 'Success',
      type: 'Letter',
    },
    {
      content: 'JVBERi0xLjUNCjQgMCBvYmoNCjw8L1R5cGUgL1BhZ2UvUGFyZW',
      encodingType: 'Pdf',
      errorMessages: { failures: null },
      erroType: null,
      id: 'abcd1234',
      loading: false,
      recipientName: 'recipient name',
      sendingMethod: 'Sms',
      status: 'Success',
      type: 'Letter',
    },
  ],
  errorText: path(['correspondence', 'errorText'], dictionary),
  labels: path(['correspondence', 'labels'], dictionary),
  onPreview: () => {},
};

const render = props =>
  shallow(<DraftCorrespondencePreviewDocuments {...defaultProps} {...props} />);

describe('<DraftCorrespondencePreviewDocuments />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the page with the default props', () => {
    const el = render();
    expect(el).toMatchSnapshot();
  });

  it('should show a Loader when generatePreviewLoading is passed in', () => {
    const el = render({ generatePreviewLoading: true });
    expect(el).toMatchSnapshot();
  });

  it('should show a fatal error if there are no docments and there is a fatal error', () => {
    const el = render({ documents: [], fatalError: true });
    expect(el).toMatchSnapshot();
  });

  it('should show a retriable error if there are no docments and there is a retriable error', () => {
    const el = render({ documents: [], retriableError: true });
    expect(el).toMatchSnapshot();
  });

  it('should be empty if documents length is 0 and no errors', () => {
    const el = render({ documents: [] });
    expect(el.children().length).toBe(0);
  });

  it('should render a loader for the documents with the loading flag', () => {
    const el = render({ documents: [{ ...defaultProps.documents[0], loading: true }] });
    expect(el.find(Loader).exists()).toBe(true);
  });

  it('should render a fatal document error for the particular document if errorType is set to FailedFatal', () => {
    const el = render({
      documents: [{ ...defaultProps.documents[0], errorType: FailedFatal, loading: true }],
    });
    expect(el).toMatchSnapshot();
  });

  it('should render a retriable document error for the particular document if errorType is set to FailedRetriable', () => {
    const el = render({
      documents: [{ ...defaultProps.documents[0], errorType: FailedRetriable, loading: true }],
    });
    expect(el).toMatchSnapshot();
  });

  it('should call forcePdfDownload with the correct content when document encoding type is pdf', () => {
    const pdfContent = 'pdf';
    const openTemplate = jest.fn();
    const el = render({
      documents: [{ ...defaultProps.documents[0], content: pdfContent }],
      onPreview: openTemplate,
    });
    el.find(sel('previewLink')).props().onClick();
    expect(openTemplate).toHaveBeenCalled();
    expect(openTemplate.mock.calls[0]).toMatchSnapshot();
    expect(forcePdfDownload).toHaveBeenCalledWith(pdfContent);
  });

  it('should open the PreviewTemplate with the relevant props when document encoding type is Html', () => {
    const htmlContent = '<p>html content</p>';
    const openTemplate = jest.fn();
    const el = render({
      documents: [{ ...defaultProps.documents[0], encodingType: HTML, content: htmlContent }],
      onPreview: openTemplate,
    });
    el.find(sel('previewLink')).props().onClick();
    expect(openTemplate).toHaveBeenCalled();
    expect(openTemplate.mock.calls[0]).toMatchSnapshot();
  });

  it('should open the PreviewTemplate with the relevant props when document encoding type is Text', () => {
    const textContent = 'asdsad asd';
    const openTemplate = jest.fn();
    const el = render({
      documents: [{ ...defaultProps.documents[0], encodingType: TEXT, content: textContent }],
      onPreview: openTemplate,
    });
    el.find(sel('previewLink')).props().onClick();
    expect(openTemplate).toHaveBeenCalled();
    expect(openTemplate.mock.calls[0]).toMatchSnapshot();
  });

  it('should add an object of documents ids to the state', () => {
    const el = render();
    expect(el.state()).toEqual({
      asdasd213123: false,
      abcd1234: false,
    });
  });

  it('should additional text once a document has been preview', () => {
    const el = render();
    el.setState({
      hours: 'yes',
      abcd1234: true,
    });

    const alreadyPreviewed = el.find('[data-bdd="document-viewed"]');
    expect(alreadyPreviewed.length).toBe(1);
  });

  it('should have helper text for a letter when isLocalPrint', () => {
    const el = render();
    el.setProps({
      isLocalPrint: true,
    });

    const manualPrintText = el.find('[data-bdd="manual-print"]');
    expect(manualPrintText.length).toBe(1);
  });

  describe('Instance methods', () => {
    let wrapper;
    let instance;

    beforeEach(() => {
      wrapper = render();
      instance = wrapper.instance();
    });

    describe('markDocumentViewed', () => {
      it('should set the state item to true', () => {
        const id = 'asdasd213123';
        expect(wrapper.state(id)).toBe(false);
        instance.markDocumentViewed(id);
        expect(wrapper.state(id)).toBe(true);
      });
    });
  });
});
