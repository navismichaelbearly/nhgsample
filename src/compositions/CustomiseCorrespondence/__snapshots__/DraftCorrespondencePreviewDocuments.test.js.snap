// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DraftCorrespondencePreviewDocuments /> should call forcePdfDownload with the correct content when document encoding type is pdf 1`] = `
Array [
  "asdasd213123",
]
`;

exports[`<DraftCorrespondencePreviewDocuments /> should open the PreviewTemplate with the relevant props when document encoding type is Html 1`] = `
Array [
  "asdasd213123",
  Object {
    "limitMaxWidth": true,
    "template": <PreviewTemplate
      content="<p>html content</p>"
      dataBddPrefix="draftCorrespondence-template"
      errorMessage="No template found, please contact your IT help desk"
      label="Close"
      onClose={[Function]}
      type="Html"
/>,
  },
]
`;

exports[`<DraftCorrespondencePreviewDocuments /> should open the PreviewTemplate with the relevant props when document encoding type is Text 1`] = `
Array [
  "asdasd213123",
  Object {
    "limitMaxWidth": true,
    "template": <PreviewTemplate
      content="asdsad asd"
      dataBddPrefix="draftCorrespondence-template"
      errorMessage="No template found, please contact your IT help desk"
      label="Close"
      onClose={[Function]}
      type="text"
/>,
  },
]
`;

exports[`<DraftCorrespondencePreviewDocuments /> should render a fatal document error for the particular document if errorType is set to FailedFatal 1`] = `
<Fragment>
  <styled.h3>
    Generated Correspondence
  </styled.h3>
  <styled.p>
    You need to view each of the following correspondence previews before proceeding.
  </styled.p>
  <p
    isSmall={false}
  />
</Fragment>
`;

exports[`<DraftCorrespondencePreviewDocuments /> should render a retriable document error for the particular document if errorType is set to FailedRetriable 1`] = `
<Fragment>
  <styled.h3>
    Generated Correspondence
  </styled.h3>
  <styled.p>
    You need to view each of the following correspondence previews before proceeding.
  </styled.p>
  <p
    isSmall={false}
  />
</Fragment>
`;

exports[`<DraftCorrespondencePreviewDocuments /> should render the page with the default props 1`] = `
<Fragment>
  <styled.h3>
    Generated Correspondence
  </styled.h3>
  <styled.p>
    You need to view each of the following correspondence previews before proceeding.
  </styled.p>
  <styled.h5>
    recipient name
     (
    Post
    )
  </styled.h5>
  <styled.div>
    <Styled(WithTheme(Button))
      data-bdd="-previewLink"
      data-state-key="previewLink"
      isText={true}
      onClick={[Function]}
    >
      View preview
    </Styled(WithTheme(Button))>
  </styled.div>
  <styled.h5>
    recipient name
     (
    Sms
    )
  </styled.h5>
  <styled.div>
    <Styled(WithTheme(Button))
      data-bdd="-previewLink"
      data-state-key="previewLink"
      isText={true}
      onClick={[Function]}
    >
      View preview
    </Styled(WithTheme(Button))>
  </styled.div>
</Fragment>
`;

exports[`<DraftCorrespondencePreviewDocuments /> should show a Loader when generatePreviewLoading is passed in 1`] = `
<p
  isSmall={false}
/>
`;

exports[`<DraftCorrespondencePreviewDocuments /> should show a fatal error if there are no docments and there is a fatal error 1`] = `
<Fragment>
  <styled.p>
    Generation failed due to DCRM error. Please contact IT Service Desk
  </styled.p>
</Fragment>
`;

exports[`<DraftCorrespondencePreviewDocuments /> should show a retriable error if there are no docments and there is a retriable error 1`] = `
<Fragment>
  <styled.p>
    Correspondence merge failed. Please retry. If the problem persists please contact IT Service Desk
  </styled.p>
</Fragment>
`;
